docs/CMakeLists.txt

```diff
@@ -12,7 +12,7 @@ add_custom_target(docs)
 
 option(ENABLE_DOXYGEN "Enable building doxygen documentation" ON)
 if (ENABLE_DOXYGEN)
-  find_package(Doxygen)
+  find_package(Doxygen REQUIRED dot)
   if (DOXYGEN_FOUND)
```

Comment:
* The `REQUIRED` option stops the configure step with an error message if `doxygen` was not found. Thus, the whole `if (DOXYGEN_FOUND)` condition is redundant as `DOXYGEN_FOUND` will always evaluate to true.
* The `dot` component of CMake's `Doxygen` package was introduced in [version 3.9](https://cmake.org/cmake/help/v3.9/module/FindDoxygen.html) but KLEE's minimal  CMake version is [3.5](https://cmake.org/cmake/help/v3.5/module/FindDoxygen.html).

Reply:
Thanks, @lzaoral .  What's the right way to do it?  Right now, things are broken too, as when Doxygen is found, `make doc-doxygen` fails as `dot` is not installed

Reply:
You mean the failing FreeBSD jobs? It seems like CMake is unable to find even the main `doxygen` binary:
```console
Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
```

However, your changes work flawlessly in my Fedora 35 VM or a fresh Ubuntu Bionic container. Both Doxygen and dot are found and `doc-doxygen` target finishes without any errors about missing `dot` executable.

As for the compatibility with CMake 3.5, setting `DOXYGEN_SKIP_DOT` to `FALSE` should be enough.

Reply:
Thanks, @lzaoral .  I was thinking more about the cmake check.  What would be the right way to code it in order to avoid the whole configure step halt if `dot` is not installed?   

For the cmake version, I think it would be reasonable to require >= 3.9, as 3.9 is quite old.

Reply:
This should work with CMake 3.9+:
```diff
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -12,7 +12,7 @@ add_custom_target(docs)

 option(ENABLE_DOXYGEN "Enable building doxygen documentation" ON)
 if (ENABLE_DOXYGEN)
-  find_package(Doxygen REQUIRED dot)
+  find_package(Doxygen REQUIRED OPTIONAL_COMPONENTS dot)
   if (DOXYGEN_FOUND)
     message(STATUS "Doxygen found")
     set(abs_top_srcdir "${CMAKE_SOURCE_DIR}")
``` 
* `doxygen` is installed but `dot` is not:
```console
$ command -v doxygen; command -v dot
/usr/bin/doxygen
$ cmake ...
...
-- Found Doxygen: /usr/bin/doxygen (found version "1.9.1") found components: doxygen missing components: dot
-- Doxygen found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/lukas/klee/build
```
Of course, `make doc-doxygen` will then produce warnings about missing `dot` executable.

As for CMake 3.5+, I misunderstood the behaviour of `DOXYGEN_SKIP_DOT` so my previous comment is wrong.  This variable controls whether we want to find `dot` or not, however, in both cases it will be an optional component.

```diff
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -12,7 +12,10 @@ add_custom_target(docs)

 option(ENABLE_DOXYGEN "Enable building doxygen documentation" ON)
 if (ENABLE_DOXYGEN)
-  find_package(Doxygen REQUIRED dot)
+  # This will not make the dot doxygen component required!  This option only
+  # controls if we want to attempt to find it.
+  set(DOXYGEN_SKIP_DOT FALSE)
+  find_package(Doxygen REQUIRED)
   if (DOXYGEN_FOUND)
     message(STATUS "Doxygen found")
     set(abs_top_srcdir "${CMAKE_SOURCE_DIR}")
```

Reply:
Btw, if you decide to switch to 3.9, you might consider trying CMake's (`doxygen_add_docs`)[https://cmake.org/cmake/help/v3.9/module/FindDoxygen.html#command:doxygen_add_docs] macro to generate a `Doxyfile` and generate the documentation instead of using a custom target that launches `doxygen` executable directly.

Reply:
You can check the [`DOXYGEN_DOT_FOUND`](https://cmake.org/cmake/help/latest/module/FindDoxygen.html#variable:DOXYGEN_DOT_FOUND) result variable and handle it accordingly.

Reply:
Got it, thanks!

Reply:
Just for completeness, if you switch to CMake 3.9, check for `Doxygen::dot` target instead as it replaced `DOXYGEN_DOT_FOUND` which is now deprecated. (Docs mention it just before the `Functions` section.)

Reply:
Thanks.  But I first noticed `DOXYGEN_HAVE_DOT`, so I was planning to use that.  It's unclear from the documentation which one is preferred.

Reply:
Yeah, CMake's documentation can be sometimes quite confusing. If I understand it correctly, the preference depends on the project's minimum required CMake version. If it's 3.5 you have no other choice than to use `DOXYGEN_HAVE_DOT` but if it's 3.9 you should check the existence of `Doxygen::dot` target instead as it had effectively replaced `DOXYGEN_HAVE_DOT`. This variable is still present in 3.9+ but only for backwards compatibility with projects that target strictly older CMake releases and shouldn't be used in projects that target newer releases.